src/inference/Inference.jl

```diff
@@ -148,8 +148,8 @@ end
 
 function AbstractMCMC.sample(
     rng::AbstractRNG,
-    model::AbstractModel,
-    sampler::Sampler{<:InferenceAlgorithm},
+    model::DynamicPPL.Model,
+    sampler::AbstractSampler,
```

Comment:
This will inevitably lead to method ambiguities though, so we need a better solution.